internal/ringbuffer.go

```diff
@@ -139,3 +135,35 @@ func (rp *RingbufferProxy) checkCounts(minCount int32, maxCount int32) (err erro
 	}
 	return
 }
+
+type LazyReadResultSet struct {
+	readCount            int32
+	dataItems            []*Data
+	itemSequences        []int64
+	serializationService *SerializationService
+}
+
+const SEQUENCE_UNAVAILABLE int64 = -1
+
+func NewLazyReadResultSet(readCount int32, items []*Data, itemSeqs []int64, ss *SerializationService) *LazyReadResultSet {
+	return &LazyReadResultSet{readCount: readCount, dataItems: items, itemSequences: itemSeqs, serializationService: ss}
+}
+
+func (rs *LazyReadResultSet) ReadCount() int32 {
+	return rs.readCount
+}
+
+func (rs *LazyReadResultSet) Get(index int32) (result interface{}, err error) {
+	return rs.serializationService.ToObject(rs.dataItems[index])
```

Comment:
Also why dont we store the object after converting the corresponding data so that when the same index is called more than once we can directly return the result. Is there any reason to avoid doing that ?

Reply:
As far as I understand it is like this in Java too,
```
    public E get(int index) {
        rangeCheck(index);
        final Data item = items[index];
        return serializationService.toObject(item);
    }
```
why is it like that ?